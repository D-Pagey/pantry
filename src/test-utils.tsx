import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

const AllTheProviders: FC = ({ children }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            }
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, options?: any) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
