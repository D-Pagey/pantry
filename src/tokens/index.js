export const CATEGORY_CARD_WIDTH = '120px';