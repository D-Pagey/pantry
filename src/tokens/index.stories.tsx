import React from 'react';
import { colours } from '.';

export default { title: 'Colours' };

export const all = () => (
    <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr 1fr 1fr 1fr', gridGap: '1rem' }}>
        {Object.entries(colours).map(([name, color]) => (
            <div key={name}>
                <p>{name}</p>
                <div style={{ width: 50, height: 50, backgroundColor: color }} />
            </div>
        ))}
    </div>
);
