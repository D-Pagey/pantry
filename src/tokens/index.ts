export const MOBILE_HEADER_HEIGHT = '80px';
export const HEADER_HEIGHT = '100px';
export const PAGE_WIDTH = '1100px';

export const CATEGORY_CARD_HEIGHT = '115px';
export const CATEGORY_CARD_WIDTH = '144px';

export const EXPIRING_SOON_DAYS = 3;

export const colours = {
    black: 'black',
    blue: 'rgb(66, 133, 244)',
    brown: 'brown',
    darkGreen10: 'rgba(52, 136, 65, 0.1)',
    darkGreen20: 'rgba(52, 136, 65, 0.2)',
    darkGreen40: 'rgba(52, 136, 65, 0.4)',
    darkGreen60: 'rgba(52, 136, 65, 0.6)',
    darkGreen100: '#348841',
    darkGrey: '#7C7C7C',
    grey: '#B2B2B2',
    lightGreen: '#31bb47',
    lightGrey: '#767676',
    veryLightGrey: '#E5E5E5',
    orange: '#FCBA3B',
    red: '#EB5757',
    white: 'white'
};

export const mediaQuery = {
    laptop: '(min-width: 1024px)',
    largeMobile: '(min-width: 375px)',
    maxPage: `(min-width: ${PAGE_WIDTH})`,
    tablet: '(min-width: 765px)'
};

export const zIndex = {
    dive: -200,
    default: 1,
    levitate: 2,
    rise: 200,
    sky: 400,
    top: 600
};
