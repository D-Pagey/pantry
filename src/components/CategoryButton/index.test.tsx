import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '../../test-utils';
import { colours } from '../../tokens';
import { CategoryButton } from '.';

const props = {
    handleClick: () => null,
    name: 'meat'
};

describe('CategoryButton component', () => {
    it('should render', () => {
        const { container } = render(<CategoryButton {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onClick when clicked', () => {
        const handleClick = jest.fn();

        render(<CategoryButton {...props} handleClick={handleClick} />);

        userEvent.click(screen.getByText('Meat'));

        expect(handleClick).toHaveBeenCalled();
    });

    it('should have selected styles if prop is selected', () => {
        render(<CategoryButton {...props} isSelected />);

        expect(screen.getByText('Meat')).toHaveStyleRule('color', colours.white);
        expect(screen.getByTestId('categoryCardmeat')).toHaveStyleRule('background-color', colours.darkGreen100);
    });
});
