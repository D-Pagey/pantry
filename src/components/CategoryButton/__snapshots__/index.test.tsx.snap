// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton component should render 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border: 1px solid rgba(52,136,65,0.6);
  border-radius: 16px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 115px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease 0s;
  transition: all 0.3s ease 0s;
  width: 144px;
}

.c1:hover {
  background-color: #348841;
}

.c2 {
  color: #348841;
  font-size: 18px;
}

.c0:hover .c2 {
  color: white;
}

<button
  class="c0 c1"
  data-testid="categoryCardmeat"
  type="button"
>
  <p
    class="c2"
  >
    Meat
  </p>
</button>
`;
